compiler/rustc_trait_selection/src/solve/eval_ctxt.rs

```diff
@@ -568,7 +569,7 @@ impl<'a, 'tcx> EvalCtxt<'a, 'tcx> {
                 GoalEvaluationKind::Nested { is_normalizes_to_hack: IsNormalizesToHack::Yes },
                 unconstrained_goal,
             )?;
-            self.add_goals(instantiate_goals);
+            self.nested_goals.goals.extend(instantiate_goals);
```

Comment:
I stopped using `add_goals` here because the added goals aren't really nested goals and it ended up breaking the proof tree builder

Reply:
the issue was that we add goals while the state is `DebugBuilder::AddedGoalsEvaluation`.

For us to sensibly handle these nested goals we'd have to either add `AddGoal` as an action to `AddedGoalsEvaluation` or somehow take the added goals during the evaluation and "move them back" into the `WipProbe` once the added goals evaluation is done. Both of these feel somewhat ugly and slightly annoying to implement